frontend_server/lib/server.dart

```diff
@@ -298,9 +291,11 @@ Future<int> starter(
           boundaryKey = string.substring(RECOMPILE_INSTRUCTION_SPACE.length);
           state = _State.RECOMPILE_LIST;
         } else if (string == 'accept') {
-          compiler.acceptLastDelta();
+          // TODO(vegorov) remove this command from the protocol
         } else if (string == 'reject') {
-          compiler.rejectLastDelta();
+          // TODO(vegorov) evaluate conditions under which deltas can be
+          // rejected by the VM.
+          compiler.resetIncrementalCompiler();
```

Comment:
With the above suggested changes to acceptLastDelta and rejectLastDelta these changes would not be needed.